src/app/map/components/EventCard/index.tsx

```diff
@@ -1,47 +1,59 @@
 "use client";
 import { Container, Flex, Text, Image, useMantineTheme } from "@mantine/core";
 import { useStyles } from "./styles";
-import type { Event } from "~/types";
+import Link from "next/link";
 import { Button } from "~/components";
 import { IconMapPin } from "@tabler/icons-react";
+import { CloseButton, Group } from "@mantine/core";
+import { EventInterface } from "../GoogleMaps";
+
 
 interface EventCardProps {
-  event: Event;
+  event: EventInterface | null;
+  onClose: () => void;
```

Comment:
I'm not sure if we should have the option to get `null` as EventInterface here. Can't we make some conditional rendering in the parent component like:
```typescript
if (events.lenght >= 1) {
    for (const event of events) {
         // logic to render infoMap
    }
}
```
Then I think we can remove all `event?`.